csv.go

```diff
@@ -0,0 +1,57 @@
+package middlewares
+
+import (
+	"bytes"
+	"encoding/json"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+func csvFromJSON(b []byte) (result []byte, err error) {
+	var data []map[string]interface{}
+	err = json.Unmarshal(b, &data)
+	if err != nil {
+		return
+	}
+	if len(data) == 0 {
+		return
+	}
+	keys := make([]string, 0, len(data[0]))
+	for k := range data[0] {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	var strOut bytes.Buffer
+	line := strings.Join(keys, ";") + "\n"
```

Comment:
You're writing csv on hand, there's the **encoding/csv**[1] package to do that job

[1] https://golang.org/pkg/encoding/csv/